pulumi/python/automation/main.py

```diff
@@ -170,19 +180,43 @@ def main():
         provider.display_execution_order(output=sys.stdout)
         sys.exit(0)
 
-    # For the other operations, we need the configuration files parsed, so we do the parsing upfront.
-
+    # We parse the environment file up front in order to have the necessary values required by this program.
+    # The logic around the PULUMI_STACK accounts for three scenarios:
+    #
+    # 1. If there is no environment file, the argument given on the CLI is used and added to the environment file.
+    # 2. If there is a difference between the CLI and the environment file, the environment file value is used.
```

Comment:
Let's error and exit in this case. This seems like the perfect recipe for disaster if someone is making assumptions.

Reply:
Where is your sense of adventure!!!

But, realistically, the chances of this blowing up is a lot higher than this serving a useful purpose. I think I was just trying to keep the behavior of the env file consistent, but in this case it makes sense to defer to throwing our hands up.